openedx_events/tooling.py

```diff
@@ -252,3 +254,31 @@ def allow_send_event_failure(self):
         More information on send_robust in the Django official documentation.
         """
         self._allow_send_event_failure = True
+
+
+def load_all_signals():
+    """
+    Ensure OpenEdxPublicSignal.all_events() cache is fully populated.
+    Loads all non-test signals.py modules.
+    """
+    found = set()
+
+    root = import_module('openedx_events')
+    for m in pkgutil.walk_packages(root.__path__, root.__name__ + '.'):
+        module_name = m.name
+        if 'tests' in module_name.split('.') or '.test_' in module_name:
+            continue
+        if module_name.endswith('.signals'):
+            import_module(module_name)
+            found.add(module_name)
+
+    # Check that the auto-discovered list matches the known modules.
+    # This is just here to ensure that the auto-discovery is working
+    # properly and doesn't start to silently fail.
+    #
+    # If this assertion fails because a module has been added, renamed,
+    # or deleted, please update the hardcoded list.
+    assert found == {
+        'openedx_events.content_authoring.signals',
+        'openedx_events.learning.signals',
+    }
```

Comment:
Part of me is a little dubious about hardcoding this list and auto-discoverying it, but I guess this is a good canary for when new signals are added.